### PR TITLE
Add Search Applications API

### PR DESCRIPTION
**Pull request by carlosdelest**
Add Search Applications API, including both CRUD and _search request.

Reopening this from https://github.com/elastic/elasticsearch-specification/pull/2080, as Github actions need the PR to be done from the same repo

### CODE REVIEW

File: specification/search_application/get/SearchApplicationsGetRequest.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * under the License.
+ */
+import { RequestBase } from '@_types/Base'
+import { Name } from '@_types/common'
+
+/**
+ * Returns the details about a search application
+ * @rest_spec_name search_application.get
+ * @since 8.8.0
+ * @stability beta
```
Comment (kderusso) [inline specification/search_application/get/SearchApplicationsGetRequest.ts:26]: I think we need to update these `@stability` tags too

**Review by kderusso — COMMENTED** [review]
Looks great so far, a couple comments! 💯

**Review by sethmlarson — APPROVED** [review]
This looks good to me!

@kderusso We'll eventually need to update the stability tags, but I was thinking we could get this PR merged and then I'll manually resync the `rest-api-spec` stubs before fixing the stabilities. Otherwise we need to resync, rebase, and all that, might be easier this way?

**Comment by carlosdelest** [comment]
@kderusso , as @sethmlarson said, this can be done in a separate PR with the syncs already done. I'm merging this PR and we'll take care of that in a separate one 👍

**Pull request merged**